src/dotnet/commands/dotnet-install/dotnet-install-tool/InstallToolCommand.cs

```diff
@@ -78,63 +80,84 @@ internal class InstallToolCommand : CommandBase
 
         public override int Execute()
         {
-            if (!_global)
+            var validateReturnCode = Validate();
+            if (validateReturnCode != 0)
             {
-                throw new GracefulException(LocalizableStrings.InstallToolCommandOnlySupportGlobal);
+                return validateReturnCode;
             }
 
-            if (_configFilePath != null && !File.Exists(_configFilePath))
-            {
-                throw new GracefulException(
-                    string.Format(
-                        LocalizableStrings.NuGetConfigurationFileDoesNotExist,
-                        Path.GetFullPath(_configFilePath)));
-            }
-
-            // Prevent installation if any version of the package is installed
-            if (_toolPackageStore.GetInstalledPackages(_packageId).FirstOrDefault() != null)
-            {
-                _errorReporter.WriteLine(string.Format(LocalizableStrings.ToolAlreadyInstalled, _packageId).Red());
-                return 1;
-            }
-
-            FilePath? configFile = null;
-            if (_configFilePath != null)
-            {
-                configFile = new FilePath(_configFilePath);
-            }
-
-            try
+            return HandleExceptionToUserFacingMessage(() =>
```

Comment:
Use action to keep all the "gut" in `Execute()`, and extract less error handling stuff to a method